plugins/inputs/mcrouter/mcrouter.go

```diff
@@ -0,0 +1,224 @@
+package mcrouter
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"net"
+	"strconv"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+// Mcrouter is a mcrouter plugin
+type Mcrouter struct {
+	Servers     []string
+	UnixSockets []string
+}
+
+var sampleConfig = `
+  ## An array of address to gather stats about. Specify an ip on hostname
+  ## with optional port. ie localhost, 10.0.0.1:11211, etc.
+  servers = ["localhost:11211"]
+  # unix_sockets = ["/var/run/mcrouter.sock"]
+`
+
+var defaultTimeout = 5 * time.Second
+
+// The list of metrics that should be sent
+var sendMetrics = []string{
```

Comment:
I would love to have these locked to specific types, since it would be a problem if a field changed values.  Maybe we can turn this into a `map[string]statType` where statType is an enum being int64, float64 or string?  Then when we parse we try the expected value and log an error if it doesn't match.

Reply:
switched to map with types and only attempt a convert for the expected type